rust/routee-compass-powertrain/src/routee/energy_traversal_model.rs

```diff
@@ -90,13 +91,58 @@ impl TraversalModel for EnergyTraversalModel {
     fn access_edge(
         &self,
         _v1: &Vertex,
-        _src: &Edge,
+        src: &Edge,
         _v2: &Vertex,
-        _dst: &Edge,
+        dst: &Edge,
         _v3: &Vertex,
-        _state: &TraversalState,
+        state: &TraversalState,
     ) -> Result<Option<TraversalState>, TraversalModelError> {
-        Ok(None)
+        match self.service.headings_table.as_deref() {
```

Comment:
this feels like a behavior that should be shared across traversal models, we'll have to think about the right abstraction there before we start adding any more of these Edge->Edge update features.

Reply:
Yeah agreed.